### DISCUSSION TITLE
Make sense to have bricks that are not python packages ?

### DISCUSSION THREAD

**Discussion by ybenitezf** [DavidVujic/python-polylith#298] [General]
Bear with me for a moment: 

- Suppose we have an ongoing project with a git repository, on the **main** branch the component `componentA` - or a base - exists. 
- In separate branch (**branchY** lest say) we make a refactor that erase  this component (some of the functions or classes are moved to other components, just to put in a real live example) and we in this brach - **branchY** - remove all references to `componentA`.
- Change to branch **main** and merge **branchY**, all good, expect that `git` will not remove the folder for `componentA`, and normally it contains some ignore files, and if we run `poly sync` then `componentA` will be added again into the root `pyproject.toml`.

So i think that this function:

https://github.com/DavidVujic/python-polylith/blob/0c7b67ad9bf91583dd6aa241d6fc6af4ef1a20db/components/polylith/bricks/component.py#L18-L19

should only return valid python packages, i. e. folder that contains a `__init__.py` directly or in a subfolder, this make sense ?

**Comment by DavidVujic** [comment]
Hi @ybenitezf!

Oh, so in the feature branch, the actual folder `componentA` isn't removed? I can imagine scenarios where there is `json` or other resource data in a component that should be included. But I agree that it would make sense to not include an empty folder in the root pyproject.toml file.

**Reply by ybenitezf** [reply]
Exactly, git do not remove the refactored component or base folder when you change branches. Maybe the removed folder only contains` __pycache__` when branches are switched - that's enough for git to not delete the folder, but  when you do `poly sync` is added to the `pyproject.toml`

**Reply by DavidVujic** [reply]
Maybe that scenario could be seen as a mistake in the feature branch? I am guessing that the folder should have been removed entirely if the code as been refactored and moved somewhere else. 

I don't want to risk that the poly tool is too restrictive about the contents of a namespace package folder, and might miss some use cases where there is other things than python code in them - and that should be part of the deploys (json, csv etc.).
